py-polars/polars/utils/_construction.py

```diff
@@ -491,35 +489,39 @@ def sequence_to_pyseries(
             # we store the values internally as UTC and set the timezone
             py_series = PySeries.new_from_anyvalues(name, values, strict)
             time_unit = getattr(dtype, "time_unit", None)
-            if time_unit is None:
+            if time_unit is None or values_dtype == Date:
                 s = wrap_s(py_series)
             else:
                 s = wrap_s(py_series).dt.cast_time_unit(time_unit)
             time_zone = getattr(dtype, "time_zone", None)
-            if dtype == Datetime and (
-                value.tzinfo is not None or time_zone is not None
-            ):
-                values_tz = str(value.tzinfo) if value.tzinfo is not None else None
-                dtype_tz = dtype.time_zone  # type: ignore[union-attr]
-                if values_tz is not None and (
-                    dtype_tz is not None and dtype_tz != "UTC"
-                ):
-                    raise ValueError(
-                        "time-zone-aware datetimes are converted to UTC"
-                        "\n\nPlease either drop the time zone from the dtype, or set it to 'UTC'."
-                        " To convert to a different time zone, please use `.dt.convert_time_zone`."
+            if dtype == Datetime:
+                if values_dtype == Date:
+                    # check for zone info
+                    return (
+                        s.cast(Datetime(time_unit)).dt.replace_time_zone(time_zone)._s
                     )
```

Comment:
is there a way to do this as an earlier return, rather than adding an extra level of indentation (to already quite indented code)?

Reply:
I could remove one level of indendation by moving it just a bit earlier with

```python
if (values_dtype == Date) and (dtype == Datetime):
    return s.cast(Datetime(time_unit)).dt.replace_time_zone(time_zone)._s
```

Which would save us 2 lines since the formatter will remove the parens. Let's try it.